crates/ruff/src/rules/pycodestyle/rules/tab_indentation.rs

```diff
@@ -16,15 +16,48 @@ impl Violation for TabIndentation {
 }
 
 /// W191
-pub fn tab_indentation(lineno: usize, line: &str) -> Option<Diagnostic> {
+pub fn tab_indentation(lineno: usize, line: &str, string_ranges: &[Range]) -> Option<Diagnostic> {
     let indent = leading_space(line);
+    if let Some(tab_index) = indent.find('\t') {
+        // If the tab character is within a multi-line string, abort.
+        if let Ok(range_index) = string_ranges.binary_search_by(|range| {
+            let start = range.location.row();
+            let end = range.end_location.row();
+            if start > lineno {
+                std::cmp::Ordering::Greater
+            } else if end < lineno {
+                std::cmp::Ordering::Less
+            } else {
+                std::cmp::Ordering::Equal
+            }
+        }) {
+            let string_range = &string_ranges[range_index];
+            let start = string_range.location;
+            let end = string_range.end_location;
+
+            // Tab is contained in the string range by virtue of lines.
+            if lineno != start.row() && lineno != end.row() {
+                return None;
+            }
+
+            let tab_column = line[..tab_index].chars().count();
+
+            // Tab on first line of the quoted range, following the quote.
+            if lineno == start.row() && tab_column > start.column() {
+                return None;
+            }
+
+            // Tab on last line of the quoted range, preceding the quote.
+            if lineno == end.row() && tab_column < end.column() {
+                return None;
+            }
```

Comment:
I think it's sufficient to just check these, and not check the "Is it on the start _and_ end line" condition, since if it's on the start _and_ end line, it'll be caught by the first condition... But hoping I'm not thinking about it incorrectly.

Reply:
Nah, I think that's right. IIRC I included a test case which contained a triple-quoted string on one line, so if that passes I think we should be good.